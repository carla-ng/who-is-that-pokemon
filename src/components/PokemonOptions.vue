<template>
    <div class="options-container">
        <ul>
            <li v-for="pokemon in pokemons"
                :key="pokemon.id"
                @click="$emit( 'pokemonSelection', pokemon.id )"
            >
                <span>{{ pokemon.name }}</span>
            </li>
        </ul>
    </div>
</template>

<script>
export default {
    props: {
        pokemons: {
            type: Array,
            required: true
        }
    }
}
</script>

<style scoped>
    ul {
        list-style-type: none;
        margin: 50px 0 0 0;
        padding: 0;
    }
    li {
        background-color: white;
        border-radius: 5px;
        border: 2px solid #000;
        cursor: pointer;
        font-weight: 600;
        letter-spacing: 2px;
        margin-bottom: 10px;
        width: 300px;
        padding: 2px;
    }

    li:hover {
        background-color: rgba(0, 0, 0, 0.05);
    }

    span {
        border: 2px solid #000;
        color: #000;
        display: block;
        font-family: Verdana, Geneva, sans-serif; 
        padding: 16px;
        text-transform: capitalize
    }

    .options-container {
        display: flex;
        justify-content: center;
    }
</style>