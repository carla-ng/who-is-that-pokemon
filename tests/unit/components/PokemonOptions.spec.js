import { shallowMount } from "@vue/test-utils"
import PokemonOptions from "@/components/PokemonOptions"
import { pokemons } from "../mocks/pokemon.mock"


describe('PokemonOptions Component', () => {

    let wrapper

    beforeEach(() => {
        wrapper = shallowMount(PokemonOptions, {
        props: {
            pokemons: pokemons,
        },
        });
    });

    
    test('Must match snapshot', () => {
    
        expect( wrapper.html() ).toMatchSnapshot();

        /*
        expect( wrapper.html() ).toMatchInlineSnapshot(`
        <div class="options-container">
            <ul>
            <li>bulbasaur</li>
            <li>ivysaur</li>
            <li>charizard</li>
            <li>blastoise</li>
            </ul>
        </div>
        `);
        */

    })


    test('Must show the 4 options correctly', () => {

        const liTags = wrapper.findAll('li')
        expect( liTags.length ).toBe(4)

        expect( liTags[0].text() ).toBe('bulbasaur')
        expect( liTags[1].text() ).toBe('ivysaur')
        expect( liTags[2].text() ).toBe('charizard')
        expect( liTags[3].text() ).toBe('blastoise')

    })


    test('Must emit "selection" with its respective params when clicked', () => {

        const [li1, li2, li3, li4] = wrapper.findAll('li')

        li1.trigger('click')
        li2.trigger('click')
        li3.trigger('click')
        li4.trigger('click')

        expect( wrapper.emitted('pokemonSelection').length ).toBe(4)
        expect( wrapper.emitted('pokemonSelection')[0] ).toEqual([1])
        expect( wrapper.emitted('pokemonSelection')[1] ).toEqual([2])
        expect( wrapper.emitted('pokemonSelection')[2] ).toEqual([6])
        expect( wrapper.emitted('pokemonSelection')[3] ).toEqual([9])

    })

});
