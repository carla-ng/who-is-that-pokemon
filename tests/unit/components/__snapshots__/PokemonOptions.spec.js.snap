// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonOptions Component Must match snapshot 1`] = `
<div class="options-container">
  <ul>
    <li>bulbasaur</li>
    <li>ivysaur</li>
    <li>charizard</li>
    <li>blastoise</li>
  </ul>
</div>
`;
